From 59e68351114a65a1f315ded1ee92f4370b8547e2 Mon Sep 17 00:00:00 2001
From: Ting Liu <b28495@freescale.com>
Date: Mon, 8 Jul 2013 17:03:43 +0800
Subject: [PATCH] fix sizeof-pointer-memaccess error

build: src/livetree.c
.../hypervisor/git-r3/git/src/devtree.c: In function 'read_intmap':
.../hypervisor/git-r3/git/src/devtree.c:1513:25: error: argument to 'sizeof' in 'memset' call is the same expression as the destination; did you mean to dereference it? [-Werror=sizeof-pointer-memaccess]
    memset(ent, 0, sizeof(ent));
                         ^
cc1: all warnings being treated as errors
build: src/ipi_doorbell.c
make[1]: *** [bin/devtree.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory `.../hypervisor/git-r3/git/output'
make: *** [all] Error 2
ERROR: oe_runmake failed

Signed-off-by: Ting Liu <b28495@freescale.com>
---
 src/devtree.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/devtree.c b/src/devtree.c
index f3710e4..3295e20 100644
--- a/src/devtree.c
+++ b/src/devtree.c
@@ -1510,7 +1510,7 @@ static void read_intmap(dt_node_t *node)
 		}
 
 		if (imap + ent->parent_naddr + ent->parent_nint > imap_end) {
-			memset(ent, 0, sizeof(ent));
+			memset(ent, 0, sizeof(intmap_entry_t));
 			break;
 		}
 
-- 
1.7.5.4

